test/sys/test_socket.rs

```diff
@@ -451,3 +451,242 @@ pub fn test_syscontrol() {
+    use nix::sys::socket::{recvmsg, sendmsg, CmsgSpace, ControlMessage, MsgFlags};
+    use nix::sys::uio::IoVec;
+    use std::io;
+    use std::io::Write;
+    use std::thread;
+
+    fn loopback_v6addr() -> Option<InterfaceAddress> {
+        let addrs = match getifaddrs() {
+            Ok(iter) => iter,
+            Err(e) => {
+                let stdioerr = io::stderr();
+                let mut handle = stdioerr.lock();
+                writeln!(handle, "getifaddrs: {:?}", e).unwrap();
+                return None;
+            },
+        };
+        for ifaddr in addrs {
+            if ifaddr.flags.contains(InterfaceFlags::IFF_LOOPBACK) {
+                match ifaddr.address {
+                    Some(SockAddr::Inet(InetAddr::V6(..))) => {
+                        return Some(ifaddr);
+                    }
+                    _ => continue,
+                }
+            }
+        }
+        None
+    }
+
+    let _m = ::LOOP_MTX
+        .lock()
+        .expect("Mutex got poisoned by another test");
+
+    let lo_ifaddr = loopback_v6addr();
+    let (lo_name, lo) = match lo_ifaddr {
+        Some(ifaddr) => (ifaddr.interface_name,
+                         ifaddr.address.expect("Expect IPv4 address on interface")),
+        None => return,
+    };
+    let receive = socket(
+        AddressFamily::Inet6,
+        SockType::Datagram,
+        SockFlag::empty(),
+        None,
+    ).expect("receive socket failed");
+    bind(receive, &lo).expect("bind failed");
+    let sa = getsockname(receive).expect("getsockname failed");
+    setsockopt(receive, Ipv6RecvPacketInfo, &true).expect("setsockopt failed");
+
+    let thread = thread::spawn(move || {
+        let mut buf = [0u8; 8];
+        let iovec = [IoVec::from_mut_slice(&mut buf)];
+        let mut space = CmsgSpace::<libc::in6_pktinfo>::new();
+        let msg = recvmsg(
+            receive,
+            &iovec,
+            Some(&mut space),
+            MsgFlags::empty(),
+        ).expect("recvmsg failed");
+        assert!(
+            !msg.flags
+                .intersects(MsgFlags::MSG_TRUNC | MsgFlags::MSG_CTRUNC)
+        );
+
+        let mut cmsgs = msg.cmsgs();
+        match cmsgs.next() {
+            Some(ControlMessage::Ipv6PacketInfo(pktinfo)) => {
+                let i = if_nametoindex(lo_name.as_bytes()).expect("if_nametoindex");
+                assert_eq!(
+                    pktinfo.ipi6_ifindex,
+                    i,
+                    "unexpected ifindex (expected {}, got {})",
+                    i,
+                    pktinfo.ipi6_ifindex
+                );
+            }
+            _ => (),
+        }
+        assert!(cmsgs.next().is_none(), "unexpected additional control msg");
+        assert_eq!(
+            iovec[0].as_slice(),
+            [1u8, 2u8, 3u8, 4u8, 5u8, 6u8, 7u8, 8u8]
+        );
+    });
+
+    let slice = [1u8, 2u8, 3u8, 4u8, 5u8, 6u8, 7u8, 8u8];
```

Comment:
Pro tip: You can leave out the "u8" on the second and subsequent elements.

Reply:
Do you want me to fix the versions above in the other existing tests or just this new one?